server/index.js

```diff
@@ -1,13 +1,36 @@
 var express = require('express');
+const mongoose = require('mongoose');
 var app = express();
 var cors = require('cors');
 const path = require('path');
 const OpenAI = require("openai");
 const OpenAIServiceFactory = require('./openaiFactory');
 
+const { MongoClient, ServerApiVersion } = require('mongodb');
+
+let pantrySchema = new mongoose.Schema({
+   PantryName: 
+   {
+      type: String, required: true, unique: true,
+   },
+   ingredients: [
+      {
+         name: {type: String, required: true},
+         category: {type: String, required: false}
+      },
+   ],
+   current: {type: Boolean, default: false}
+});
+
+const Pantry = mongoose.model('Pantry', pantrySchema);
+
+
+mongoose.connect("MONGOBD_URL", { useNewUrlParser: true, useUnifiedTopology: true })
```

Comment:
Is this where the provided MongoDB url go into?

Reply:
Confirmed yes.

Reply:
Yes.